packages/gatsby-cli/src/reporter/index.js

```diff
@@ -75,43 +65,25 @@ module.exports = Object.assign(reporter, {
   uptime(prefix: string) {
     this.verbose(`${prefix}: ${(process.uptime() * 1000).toFixed(3)}ms`)
   },
+  success: (...args) => inkReporter.onSuccess(...args),
+  verbose: (...args) => inkReporter.onVerbose(...args),
+  info: (...args) => inkReporter.onInfo(...args),
+  warn: (...args) => inkReporter.onWarn(...args),
+  log: (...args) => inkReporter.onLog(...args),
   /**
    * Time an activity.
    * @param {string} name - Name of activity.
    * @param {activityArgs} activityArgs - optional object with tracer parentSpan
    * @returns {string} The elapsed time of activity.
    */
   activityTimer(name, activityArgs: ActivityArgs = {}) {
-    const spinner = reporter.activity()
-    const start = process.hrtime()
-    let status
-
-    const elapsedTime = () => {
-      var elapsed = process.hrtime(start)
-      return `${convertHrtime(elapsed)[`seconds`].toFixed(3)} s`
-    }
+    return inkReporter.createActivity(name, activityArgs)
+  },
+}
 
-    const { parentSpan } = activityArgs
-    const spanArgs = parentSpan ? { childOf: parentSpan } : {}
-    const span = tracer.startSpan(name, spanArgs)
+console.log = reporter.log
```

Comment:
We should _make sure_ we want to overwrite globals like this. It does solve the problem nicely re: plugins opting in to our reporter, though 🤔

Reply:
I'm not the biggest fan of this either but INK doesn't play super nicely with other process writing to stdout.

I've looked into the node codebase and they are actually just using util.format so I'm using it too.
https://github.com/gatsbyjs/gatsby/pull/13089/files#diff-5753f39bea69d2159152a45abdca82eaR87

I probably want to add some tests (maybe react snapshots could work here) to see if everything is wired correctly.

Reply:
yeah, my thinking is that instead of directly replacing console.log with reporter.log we should `util.format` what's passed in and then send it to reporter.log — we're going to overload the reporters so that core & plugins can send in structured data which can then be rendered by a component for richer reporting. So we can replace all the places in core where we're trying to do more sophisticated layout of messages with sending data to a component.

Reply:
we'll centralize design in some standard components which we can continue to iterate on over time + we'll have structured data that a) we can send out as JSON for `--json` output and that we can send to other UIs e.g. the coming Gatsby UI

Reply:
I think if we look at structured logging so have multiple ways of saving these logs I think it's better to move to winston and have an ink reporter as a plugin. (probably v3)

i'll like the console.log switch, let me update that.